File: backend/internal/models/product_category.go
// internal/models/product_category.go
package models

type ProductCategory struct {
	ID   uint   `gorm:"primaryKey" json:"id"`
	Name string `gorm:"not null" json:"name"`
	Slug string `gorm:"uniqueIndex;not null" json:"slug"`
}

func (ProductCategory) TableName() string {
	return "product_categories"
}

File: web-app/agrohub_mobile/farmer_mobile/lib/features/sell/models/product_category.dart
// lib/features/sell/models/product_category.dart
class ProductCategory {
  final int id;
  final String name;
  final String slug;
  final String? icon;

  ProductCategory({
    required this.id,
    required this.name,
    required this.slug,
    this.icon,
  });

  factory ProductCategory.fromJson(Map<String, dynamic> json) {
    return ProductCategory(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      slug: json['slug'] ?? '',
      icon: json['icon'],
    );
  }
}
